doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,375 @@
+```
+
+The user defines `someAsyncApiCall()` to have an asynchronous signature,
+actually operates synchronously.  When it is called, the callback provided to
+`someAsyncApiCall ()` is called in the same phase of the event loop because
+`someAsyncApiCall()` doesn't actually do anything asynchronously.  As a
+result, the callback tries to reference `bar` but it may not have that
+variable in scope yet because the script has not been able to run to
+completion.
+
+By placing it in a `process.nextTick()`, the script still has the ability to
+run to completion, allowing all the variables, functions, etc., to be
+initialized prior to the callback being called.   It also has the advantage of
+not allowing the event loop to continue.  It may be useful that the user be
+alerted to an error before the event loop is allowed to continue.
+
+A real world example in node would be:
+
+```js
+const server = net.createServer(() => {}).listen(8080);
+
+server.on('listening', () => {});
+```
+
+When only a port is passed the port is bound immediately. So the `'listening'`
+callback could be called immediately. Problem is that the `.on
+('listening')` will
+not have been set by that time.
+
+To get around this the `'listening'` event is queued in a `nextTick()` to allow
+the script to run to completion. Which allows the user to set any event
+handlers they want.
+
+## process.nextTick() vs `setImmediate()`
+
+We have two calls that are similar as far as users are concerned, but their
+names are confusing.
+
+* `process.nextTick()` fires immediately on the same phase
+* `setImmediate()` fires on the following iteration or 'tick' of the event
+  loop
+
+In essence, the names should be swapped.  `process.nextTick()` fires more
+immediately than `setImmediate()` but this is an artifact of the past which is
+unlikely to change. Making this switch would break a large percentage of the
+packages on npm. Every day more new modules are being added, which mean every
+day we wait, more potential breakages occur.  While they are confusing, the
+names themselves won't change.
+
+*We recommend developers use `setImmediate()` in all cases because its easier
+to reason about (and it leads to code that's compatible with a wider
+variety of environments, like browser JS.)*
+
+## Why use `process.nextTick()`? 
+
+There are two main reasons:
+
+1. Allow users to handle errors, cleanup any then unneeded resources, or
+perhaps try the request again before the event loop continues.
+
+2. At times it's necessary to allow a callback to run after the call
+stack has unwound but before the event loop continues. 
+
+One example is to match the user's expectations. Simple example:
+
+```js
+var server = net.createServer();
+server.on('connection', function(conn) { });
+
+server.listen(8080);
+server.on('listening', function() { });
+```
+
+Say that listen() is run at the beginning of the event loop, but the 
+listening callback is placed in a setImmediate. Now, unless a hostname 
```

Comment:
All instances of `setImmediate` should probably be styled like:

<pre>
`setImmediate()`
</pre>